File: iOS SoonZik/SoonZik/Classes/Pack.m
//
//  Pack.m
//  SoonZik
//
//  Created by Maxime Sauvage on 12/09/2014.
//  Copyright (c) 2014 Coordina. All rights reserved.
//

#import "Pack.h"

@implementation Pack

/* @property (nonatomic, assign) int identifier;
 @property (nonatomic, strong) NSString *title;
 @property (nonatomic, strong) Genre *genre;
 @property (nonatomic, assign) float price;
 @property (nonatomic, strong) NSArray *listOfAlbums;
 @property (nonatomic, strong) NSArray *listOfComments;*/

- (id)initWithJsonObject:(NSDictionary *)json
{
    self = [super init];
    self.identifier = [[json objectForKey:@"id"] intValue];
    self.title = [json objectForKey:@"title"];
    self.price = [[json objectForKey:@"price"] floatValue];
    self.listOfAlbums = [[NSMutableArray alloc] init];
    NSDictionary *albums = [json objectForKey:@"albums"];
    for (NSDictionary *album in albums) {
        //NSLog(@"%@", album);
        Album *al = [[Album alloc] initWithJsonObject:album];
        [self.listOfAlbums addObject:al];
    }
    
    return self;
}

@end

File: Android/TestSuperModel/app/src/main/java/com/soonzik/testsupermodel/Pack.java
package com.soonzik.testsupermodel;

import org.json.JSONObject;

import java.util.ArrayList;

/**
 * Created by kevin_000 on 21/03/2015.
 */
public class Pack extends ActiveRecord {
    private int id = -1;
    private String title = "";
    private Genre genre = null;
    private ArrayList<Album> albums = null;


    public Pack() {}

    public Pack(JSONObject json) {
        super.createInstance(this, json, this.getClass());
    }

    @Override
    public String toString() {
        String str = "id = " + Integer.toString(id)
                + " : title = " + title
                + " : Genre = {" + (genre != null ? genre.toString() : "") + " }";
        str += " : Album = [ ";
        if (albums != null) {
            for (Album album: albums) {
                str += "{ " + album.toString() + " } ";
            }
        }
        str += " ]";
        return (str);
    }

    public String getTitle() {
        return title;
    }

    public int getId() {
        return id;
    }
}
